### PR TITLE
Feat: Customize API base URL

### PR DESCRIPTION
**Pull request by hanjano**

### CODE REVIEW

File: blockapi/v2/base.py [hunk 1]
```diff
@@ -79,6 +79,18 @@ def __repr__(self):
         return f"{self.__class__.__name__}(coin={self.coin.name})"
 
 
+class CustomizableBlockchainApi(BlockchainApi, ABC):
+    """
+    Class for handling blockchain API services with customizable base URL.
+    """
```
Comment (galvanizze) [inline blockapi/v2/base.py:85]: please add example of common cases (RPC services, testnet alternatives, ...)

File: blockapi/test/v2/api/test_solana.py [hunk 2]
```diff
@@ -325,3 +325,13 @@ def test_get_balance(solana_api):
...
+
+
+def test_use_custom_url():
+    api = SolanaApi('https://proxy/solana/')
+    assert api.api_options.base_url == 'https://proxy/solana/'
+
+
+def test_use_base_url():
+    api = SolanaApi()
+    assert api.api_options.base_url == 'https://api.mainnet-beta.solana.com/'
```
Comment (galvanizze) [inline blockapi/test/v2/api/test_solana.py:337]:
```suggestion
    assert api.api_options.base_url == api.API_BASE_URL
```

**Review by galvanizze — APPROVED** [review]

File: blockapi/v2/base.py [hunk 3]
```diff
@@ -79,6 +79,19 @@ def __repr__(self):
         return f"{self.__class__.__name__}(coin={self.coin.name})"
 
 
+class CustomizableBlockchainApi(BlockchainApi, ABC):
+    """
+    Class for handling blockchain API services with customizable base URL,
+    e.g. proxy, testnet, RPC services, alternative sources
+    """
+
+    API_BASE_URL: str = NotImplemented
```
Comment (ladovina) [inline blockapi/v2/base.py:88]: nitpick: I though this is wrong so I investigated what happens when you use just `NotImplemented`, looks it works fine (I thought the var will just contain `<class NotImplemented>` (which is not)... it was not clear to me in the official doc: https://docs.python.org/3/library/constants.html#NotImplemented  but this was better: https://s16h.medium.com/pythons-notimplemented-type-2d720137bf41

tldr: probably use of property with `raise NotImplemented` would be probably better

**Review by ladovina — APPROVED** [review]
